fw/rbcx-coprocessor/include/Motor.hpp

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "utils/Debug.hpp"
+#include "utils/Regulator.hpp"
+
+#include <algorithm>
+#include <stdint.h>
+
+inline const uint16_t motorLoopFreq = 100;
+
+enum class MotorState {
+    PowerCtrl,
+    BrakeCtrl,
+    VelocityCtrl,
+};
+
+class Motor {
+public:
+    Motor()
+        : m_velocityReg(SHRT_MAX, 150000, 300000, 20000)
+        , m_state(MotorState::PowerCtrl) {}
+
+    std::pair<int16_t, bool> poll(uint16_t encTicks) {
+        int16_t actualTicksPerLoop = encTicks - m_lastEncTicks;
+        m_lastEncTicks = encTicks;
+        if (m_state == MotorState::PowerCtrl) {
+            return std::make_pair(m_targetPower, false);
+        } else if (m_state == MotorState::BrakeCtrl) {
+            return std::make_pair(m_targetBrakingPower, true);
+        } else if (m_state == MotorState::VelocityCtrl) {
+            int16_t targetTicksPerLoop = m_targetVelocity / motorLoopFreq;
+            int16_t targetTicksRem = abs(m_targetVelocity % motorLoopFreq);
+            int16_t dither = abs(m_dither % motorLoopFreq);
```

Comment:
m_dither bude vždy v rozmezí <0, 3>, je tohle potřeba?

Reply:
Samozřejmě, je to pitomost